model/artifact/artifact_file.go

```diff
@@ -85,6 +93,18 @@ func GetAllArtifacts(tasks []TaskIDAndExecution) ([]File, error) {
 	}
 	files := []File{}
 	for _, artifact := range artifacts {
+		for i := range artifact.Files {
+			if artifact.Files[i].Visibility == "signed" {
+				if artifact.Files[i].AwsSecret == "" || artifact.Files[i].AwsKey == "" || artifact.Files[i].Bucket == "" || artifact.Files[i].FileKey == "" {
+					return nil, errors.Errorf("error presigning the url for %s, awsSecret, awsKey, bucket, or filekey missing", artifact.Files[i].Name)
+				}
+				urlStr, err := thirdparty.PreSign(artifact.Files[i].Bucket, artifact.Files[i].FileKey, artifact.Files[i].AwsKey, artifact.Files[i].AwsSecret)
+				if err != nil {
```

Comment:
would a viable alternative in the case of an error be to skip signing this file?

Reply:
resolved in person

Reply:
discussed in person